cub/cub/device/device_segmented_reduce.cuh

```diff
@@ -208,35 +273,36 @@ struct DeviceSegmentedReduce
             typename EndOffsetIteratorT,
             typename ReductionOpT,
             typename T>
-  CUB_RUNTIME_FUNCTION static cudaError_t Reduce(void *d_temp_storage,
-                                                 size_t &temp_storage_bytes,
-                                                 InputIteratorT d_in,
-                                                 OutputIteratorT d_out,
-                                                 int num_segments,
-                                                 BeginOffsetIteratorT d_begin_offsets,
-                                                 EndOffsetIteratorT d_end_offsets,
-                                                 ReductionOpT reduction_op,
-                                                 T initial_value,
-                                                 cudaStream_t stream = 0)
+  CUB_RUNTIME_FUNCTION static cudaError_t Reduce(
+    void* d_temp_storage,
+    size_t& temp_storage_bytes,
+    InputIteratorT d_in,
+    OutputIteratorT d_out,
+    int num_segments,
+    BeginOffsetIteratorT d_begin_offsets,
+    EndOffsetIteratorT d_end_offsets,
+    ReductionOpT reduction_op,
+    T initial_value,
+    cudaStream_t stream = nullptr)
   {
     // Integer type for global offsets
     using OffsetT = detail::common_iterator_value_t<BeginOffsetIteratorT, EndOffsetIteratorT>;
+    using integral_offset_check = ::cuda::std::is_integral<OffsetT>;
 
-    return DispatchSegmentedReduce<InputIteratorT,
-                                   OutputIteratorT,
-                                   BeginOffsetIteratorT,
-                                   EndOffsetIteratorT,
-                                   OffsetT,
-                                   ReductionOpT>::Dispatch(d_temp_storage,
-                                                           temp_storage_bytes,
-                                                           d_in,
-                                                           d_out,
-                                                           num_segments,
-                                                           d_begin_offsets,
-                                                           d_end_offsets,
-                                                           reduction_op,
-                                                           initial_value,
-                                                           stream);
+    static_assert(integral_offset_check::value, "Offset iterator type should be integral.");
```

Comment:
suggestion: I'm not sure about this wording. The offset iterator type can be anything, starting from a `int*`, ending with `thrust::counting_iterator`. Is think it should be something along the lines of `Offset type should be integral`, or `Offset iterator value type should be integral`.